python/arcticdb/_msgpack_compat.py

```diff
@@ -4,15 +4,54 @@
 
 This module implements a backwards compatible version of msgpack functions.
 """
-
 import msgpack
+from arcticdb.log import version as log
+from arcticdb.exceptions import ArcticNativeException
+
+ExtType = msgpack.ExtType
+
+
+def _check_valid_msgpack():
+    pack_module = msgpack.packb.__module__
+    packer_module = msgpack.Packer.__module__
+    # Check that msgpack hasn't been monkey patched by another package
+    # We only support the official cmsgpack and fallback modules
+    if (pack_module in ("msgpack", "msgpack.fallback")) and (
+        packer_module in ("msgpack._packer", "msgpack.fallback", "msgpack._cmsgpack")
+    ):
+        return
+    log.info("Unsupported msgpack variant, got: {}, {}".format(pack_module, packer_module))
+    raise ArcticNativeException("Unsupported msgpack variant, got: {}, {}".format(pack_module, packer_module))
+
 
+def packb(obj, **kwargs):
+    _check_valid_msgpack()
+    # use_bin_type supported from msgpack==0.4.0 but became true later
+    return msgpack.packb(obj, use_bin_type=True, strict_types=True, **kwargs)
 
-def unpackb(*args, **kwargs):
+packb.__doc__ = msgpack.packb.__doc__
+packb.__name__ = msgpack.packb.__name__
+
+
+def padded_packb(obj, **kwargs):
+    """msgpack.packb with some defaults across msgpack versions and padded to 8 bytes
+    returns: (packed bytes, nbytes of unpadded content)"""
+    _check_valid_msgpack()
+    # use_bin_type is supported from msgpack==0.4.0 but became true later
+    # don't reset the buffer so we can append padding bytes
+    packer = msgpack.Packer(autoreset=False, use_bin_type=True, strict_types=True, **kwargs)
+    packer.pack(obj)
+    nbytes = packer.getbuffer().nbytes
+    pad = -nbytes % 8 # next multiple of 8 bytes
+    [packer.pack(None) for _ in range(pad)] # None is packed as single byte b`\xc0`
```

Comment:
`None is packed as single byte b"\xc0"`
As we're relying on this assumption, can we add a tiny test that it is the case so that we notice early if we upgrade `msgpack` and it changes?

Reply:
done

Reply:
done